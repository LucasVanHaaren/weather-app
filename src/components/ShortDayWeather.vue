<template>
  <v-card class="mx-auto" max-width="400px" outlined >
    <v-list-item>
      <v-list-item-content>
        <v-list-item-title class="headline mb-1">
          <b>{{ name }}</b>
        </v-list-item-title>
        <v-list-item>
          <v-list-item-icon>
            <v-icon>fas fa-thermometer-quarter</v-icon>
          </v-list-item-icon>
          <v-list-item-title>
            {{ temperature }} Â°C
          </v-list-item-title>
        </v-list-item>
        <v-list-item>
          <v-list-item-icon>
            <v-icon>fas fa-tint</v-icon>
          </v-list-item-icon>
          <v-list-item-title>
            {{ humidity }} %
          </v-list-item-title>
        </v-list-item>
      </v-list-item-content>
      <v-list-item-avatar tile size="100">
        <img :src="icon_url" />
      </v-list-item-avatar>
    </v-list-item>
  </v-card>
</template>

<script>
  export default {
    name: 'ShortDayWeather',
    props: {
      name: String,
      temperature: Number,
      humidity: Number,
      icon_url: String,
    }
  }
</script>
