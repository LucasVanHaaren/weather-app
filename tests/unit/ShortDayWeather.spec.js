import { shallowMount } from '@vue/test-utils'
import ShortDayWeather from "@/components/ShortDayWeather.vue"

describe('ShortDayWeather.vue', () => {
  it('match snapshot', () => {
    const wrapper = shallowMount(ShortDayWeather, {
      propsData: {
        name: "Monday",
        temperature: 22,
        humidity: 45,
        icon_url: "https://openweathermap.org/img/wn/01d@4x.png",
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
