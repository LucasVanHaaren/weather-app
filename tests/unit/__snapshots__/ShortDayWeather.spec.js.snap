// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortDayWeather.vue match snapshot 1`] = `
<v-card max-width="400px" outlined="" class="mx-auto">
  <v-list-item>
    <v-list-item-content>
      <v-list-item-title class="headline mb-1"><b>Monday</b></v-list-item-title>
      <v-list-item>
        <v-list-item-icon>
          <v-icon>fas fa-thermometer-quarter</v-icon>
        </v-list-item-icon>
        <v-list-item-title>
          22 Â°C
        </v-list-item-title>
      </v-list-item>
      <v-list-item>
        <v-list-item-icon>
          <v-icon>fas fa-tint</v-icon>
        </v-list-item-icon>
        <v-list-item-title>
          45 %
        </v-list-item-title>
      </v-list-item>
    </v-list-item-content>
    <v-list-item-avatar tile="" size="100"><img src="https://openweathermap.org/img/wn/01d@4x.png"></v-list-item-avatar>
  </v-list-item>
</v-card>
`;
